tensorflow/lite/delegates/webnn/webnn_delegate.cc

```diff
@@ -350,37 +337,46 @@ class Subgraph {
     }
 
     if (any_pointers_changed) {
-      graph_inputs_ = wnn::CreateNamedInputs();
+      graph_inputs_ = emscripten::val::object();
       for (int t : inputs_) {
-        wnn_inputs_[t].resource.arrayBufferView.buffer = context->tensors[t].data.raw;
-        wnn_inputs_[t].resource.arrayBufferView.byteLength = context->tensors[t].bytes;
         std::string name = std::to_string(t);
-        graph_inputs_.Set(name.c_str(), &wnn_inputs_[t]);
+        auto input_size = context->tensors[t].bytes / 4;
+        auto input_data = context->tensors[t].data.f;
+        emscripten::val view{ emscripten::typed_memory_view(input_size, input_data) };
+        auto input = emscripten::val::global("Float32Array").new_(input_size);
+        input.call<void>("set", view);
+        graph_inputs_.set(name, input);
       }
 
-      graph_outputs_ = wnn::CreateNamedOutputs();
+      graph_outputs_ = emscripten::val::object();
       for (int t : outputs_) {
-        wnn_outputs_[t].arrayBufferView.buffer = context->tensors[t].data.raw;
-        wnn_outputs_[t].arrayBufferView.byteLength = context->tensors[t].bytes;
         std::string name = std::to_string(t);
-        graph_outputs_.Set(name.c_str(), &wnn_outputs_[t]);
+        auto output_size = context->tensors[t].bytes / 4;
+        auto output = emscripten::val::global("Float32Array").new_(output_size);
+        graph_outputs_.set(name, output);
```

Comment:
ditto, you probably could use `emscripten::typed_memory_view` to create a `Float32Array` from `context->tensors[t].bytes / 4` and `context->tensors[t].data.f`, and set it to `graph_outputs_`.